Detectors/DCS/src/DCSProcessor.cxx

```diff
@@ -0,0 +1,634 @@
+      mAliaseschars.emplace_back((*it).get_alias(), DeliveryType::RAW_CHAR);
+      nchars++;
+    }
+    if ((*it).get_type() == DeliveryType::RAW_INT) {
+      mAliasesints.emplace_back((*it).get_alias(), DeliveryType::RAW_INT);
+      nints++;
+    }
+    if ((*it).get_type() == DeliveryType::RAW_DOUBLE) {
+      mAliasesdoubles.emplace_back((*it).get_alias(), DeliveryType::RAW_DOUBLE);
+      ndoubles++;
+    }
+    if ((*it).get_type() == DeliveryType::RAW_UINT) {
+      mAliasesUints.emplace_back((*it).get_alias(), DeliveryType::RAW_UINT);
+      nUints++;
+    }
+    if ((*it).get_type() == DeliveryType::RAW_BOOL) {
+      mAliasesbools.emplace_back((*it).get_alias(), DeliveryType::RAW_BOOL);
+      nbools++;
+    }
+    if ((*it).get_type() == DeliveryType::RAW_STRING) {
+      mAliasesstrings.emplace_back((*it).get_alias(), DeliveryType::RAW_STRING);
+      nstrings++;
+    }
+    if ((*it).get_type() == DeliveryType::RAW_TIME) {
+      mAliasestimes.emplace_back((*it).get_alias(), DeliveryType::RAW_TIME);
+      ntimes++;
+    }
+    if ((*it).get_type() == DeliveryType::RAW_BINARY) {
+      mAliasesbinaries.emplace_back((*it).get_alias(), DeliveryType::RAW_BINARY);
+      nbinaries++;
+    }
+  }
+
+  mLatestTimestampchars.resize(nchars);
+  mLatestTimestampints.resize(nints);
+  mLatestTimestampdoubles.resize(ndoubles);
+  mLatestTimestampUints.resize(nUints);
+  mLatestTimestampbools.resize(nbools);
+  mLatestTimestampstrings.resize(nstrings);
+  mLatestTimestamptimes.resize(ntimes);
+  mLatestTimestampbinaries.resize(nbinaries);
+
+  for (auto i = 0; i < nchars; i++) {
+    mLatestTimestampchars[i] = 0;
+  }
+  for (auto i = 0; i < nints; i++) {
+    mLatestTimestampints[i] = 0;
+  }
+  for (auto i = 0; i < ndoubles; i++) {
+    mLatestTimestampdoubles[i] = 0;
+  }
+  for (auto i = 0; i < nUints; i++) {
+    mLatestTimestampUints[i] = 0;
+  }
+  for (auto i = 0; i < nbools; i++) {
+    mLatestTimestampbools[i] = 0;
+  }
+  for (auto i = 0; i < nstrings; i++) {
+    mLatestTimestampstrings[i] = 0;
+  }
+  for (auto i = 0; i < ntimes; i++) {
+    mLatestTimestamptimes[i] = 0;
+  }
+  for (auto i = 0; i < nbinaries; i++) {
+    mLatestTimestampbinaries[i] = 0;
+  }
+}
+
+//__________________________________________________________________
+
+int DCSProcessor::process(const std::unordered_map<DPID, DPVAL>& map)
+{
+
+  // process function to do "something" with the DCS map that is passed
+
+  // first, we need to check if there are the Data Points that we need
+
+  auto foundChars = 0, foundInts = 0, foundDoubles = 0, foundUInts = 0,
+       foundBools = 0, foundStrings = 0, foundTimes = 0, foundBinaries = 0;
+
+  // char type
+  auto s = mAliaseschars.size();
+  if (s > 0) {
+    for (size_t i = 0; i != mAliaseschars.size(); ++i) {
+      int count = 0;
+      std::unordered_map<DPID, DPVAL>::const_iterator it;
+      processAlias(mAliaseschars[i], DeliveryType::RAW_CHAR, map, count, it);
```

Comment:
Indeed I can check the validity where I call "processAlias", I did not see it since I was taking a piece of code and making it a separate function.
As for return object vs argument, what is the difference? I mean, they do the same job. Is one better?
As for using auto, I would explicitly mention the type for clarity.

Reply:
BTW, I cannot return it because it is a local variable. This is why I did it in the other way. Am I wrong?

Reply:
Ok, i thought it would have been copied, which is why I would have used a reference, but one cannot use it to a local variable. Since it is moved, it is of course fine.